pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,572 @@
 title: Slice grammar
 ---
 
+This page describes the grammatical rules, usage, and restrictions of the Slice language.
+
+## Slice elements
+
+### Slice files
+
+The `SliceFile` grammar rule is the parser's entry point.
+Once a file's contents have been [preprocessed], the resulting text is matched against this rule, in its entirety.
+
+While the contents of a Slice file may reference things defined in other files, each file is parsed separately.
+There is no notion of 'including' the contents of one file in another, and errors in one file do not affect the parsing of others.
+
+A Slice file consists of an optional [module declaration][module-declaration], followed by any number of Slice definitions.
+Additionally, [file attributes][attribute] and a [mode declaration][mode-declaration] may be specified at the top of a file, before any other definitions.
+Declaring more than one mode in a single file is forbidden.
+
+```ebnf {% showTitle=false %}
+SliceFile
+    : SliceFilePrelude Module? Definition*
+    ;
+
+SliceFilePrelude
+    : EMPTY
+    | SliceFilePrelude FileCompilationMode
+    | SliceFilePrelude FileAttribute
+    ;
+
+Definition
+    : Struct | Class | Exception | Interface | Enum | CustomType | TypeAlias
+    ;
+```
+
+For additional information on Slice files, see the [slice file][slice-file-guide] page.
+
+### Mode declarations
+
+Mode declarations consist of the `mode` keyword, followed by an equals sign, followed by an identifier.
+That identifier must be a valid Slice compilation mode - either `Slice1` or `Slice2`.
+
+```ebnf {% showTitle=false %}
+FileCompilationMode
+    : "mode" "=" Identifier
+    ;
+```
+
+Mode declarations must come before any Slice definitions, including [module declarations][module-declaration].
+While the compilation mode you select will affect which Slice features are available to you,
+it has no effect on the parsing of, or validity of syntax within, your Slice file.
+For example, even though classes are a Slice1-only feature, `class` is still a keyword within a Slice2 file,
+and if you define a class, syntax errors in that class's definition will still be reported normally.
```

Comment:
I think emphasizing that we're talking about that class is important here.

Otherwise the phrase could mean: "we report a syntax error for using a class",
when what it's conveying is: "we still fully parse the class for syntax errors".